tests/semantic_analyser.cpp

```diff
@@ -1436,6 +1436,71 @@ TEST(semantic_analyser, builtin_args)
   test(*bpftrace, "t:sched:sched_one { args->not_a_field }", 1);
 }
 
+TEST(semantic_analyser, type_ctx)
+{
```

Comment:
would be good to test some invalid constructs here too

Reply:
I added https://github.com/iovisor/bpftrace/pull/1104/commits/56184dd61acce7722cf44536c9a06f9de0d4f6f4#diff-841d87c729a5da4754074bc000917a99